dracclient/resources/raid.py

```diff
@@ -98,15 +98,229 @@
+                RAID attribute
+        :param min_length: minimum length of the string
+        :param max_length: maximum length of the string
+        """
+        super(RAIDStringAttribute, self).__init__(name, instance_id,
+                                                  pending_value,
+                                                  read_only, fqdd)
+        self.current_value = current_value
+        self.min_length = min_length
+        self.max_length = max_length
+
+    @classmethod
+    def parse(cls, raid_attr_xml):
+        """Parses XML and creates RAIDStringAttribute object"""
+
+        raid_attr = RAIDAttribute.parse(cls.namespace, raid_attr_xml)
+        current_value = [attr.text for attr
+                         in utils.find_xml(raid_attr_xml,
+                                           'CurrentValue',
+                                           cls.namespace, find_all=True)]
+        min_length = int(utils.get_wsman_resource_attr(
+            raid_attr_xml, cls.namespace, 'MinLength'))
+        max_length = int(utils.get_wsman_resource_attr(
+            raid_attr_xml, cls.namespace, 'MaxLength'))
+
+        return cls(raid_attr.name, raid_attr.instance_id,
+                   raid_attr.pending_value, raid_attr.read_only,
+                   raid_attr.fqdd, current_value, min_length, max_length)
+
+
 class RAIDManagement(object):
 
+    NAMESPACES = [(uris.DCIM_RAIDEnumeration, RAIDEnumerableAttribute),
+                  (uris.DCIM_RAIDString, RAIDStringAttribute)]
+
     def __init__(self, client):
         """Creates RAIDManagement object
 
         :param client: an instance of WSManClient
         """
         self.client = client
 
+    def list_raid_controller_settings(self, by_name=False):
+        """List the RAID configuration settings
+
+        :param by_name: Controls whether returned dictionary uses RAID
+                        attribute name or instance_id as key.
+        :returns: a dictionary with the RAID settings using InstanceID as the
+                  key. The attributes are either RAIDEnumerableAttribute,
+                  RAIDStringAttribute objects.
+        :raises: WSManRequestFailure on request failures
+        :raises: WSManInvalidResponse when receiving invalid response
+        :raises: DRACOperationFailed on error reported back by the DRAC
+                interface
+        """
+
+        return utils.list_settings(self.client, self.NAMESPACES, by_name)
+
+    def set_raid_controller_settings(self, new_settings, raid_fqdd):
+        """Sets the RAID configuration
+
+        It sets the pending_value parameter for each of the attributes
+        passed in. For the values to be applied, a config job must
+        be created.
+        :param new_settings: a dictionary containing the proposed values, with
+                         each key being the name of attribute and the value
+                         being the proposed value.
```

Comment:
I'm wondering if we have a significant bug here.  Looking at DCIM_RAIDEnumeration, there are a lot of attributes that have the same AttributeName, but different InstanceIDs.  In utils.set_settings, it calls list_settings(by_name=True).  This will result in a much smaller list of attributes than if it had been called with by_name=False, which will make everything based off current_settings behave strangely.
If a user passed in RAIDdefaultWritePolicy for an attribute name, what would the iDRAC do?  I would think it would have to error out since it couldn't determine what attribute to set.  What if the user wanted to set that attribute on 2 different virtual disks?  How would they do that?  Would they pass in the InstanceID instead of the name?  If they did that, then wouldn't set_settings fail because it wouldn't find that attribute?  Is it valid to put the InstanceID in the SetAttributes packet instead of the attribute name?

Reply:
Yes right, for many of the attributes AttributeName is same ,so iDRAC will not be able to find the corresponding attribute which need to be set.
I will try to put InstanceID in SetAtrributes for RAID attributes and test those changes. And will update the PR accordingly.